### DISCUSSION TITLE
socketio.AsyncServer doesn't cancel background tasks

### DISCUSSION THREAD

**Discussion by nsmkhn** [miguelgrinberg/python-socketio#851] [Q&A] [answered]
hi @miguelgrinberg 

AsyncServer deployed with Sanic reports the following log message when app is shutdown with <Ctrl-C>: Task was destroyed but it is pending!
task: <Task pending coro=<"task_name" running at "filename"> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x10bbcc6d8>()]>>

Shouldn't AsyncServer cancel / stop background tasks gracefully when app is shut?

**Comment by miguelgrinberg** [comment, marked as answer]
Which task is giving you this problem? Is the task really called `task_name`? I don't recognize that as being part of this package.

**Reply by nsmkhn** [reply]
Here is the task which causes this:

```python
async def __gather_and_send_events(self):
    while True:
        await self.sio.sleep(self.listener_timeout)
        for event in self.event_listener.get_events():
            for topic in event.topics:
                await self.sio.emit('event', event.data, room=topic)

def __register_app_listeners(self):
    @self.app.listener('after_server_start')
    def __start_event_listener_task(app, loop):
        self.sio.start_background_task(self.__gather_and_send_events)
```

**Reply by miguelgrinberg** [reply]
This is your own task, it is your responsibility to stop them gracefully when the server is stopped.

**Reply by nsmkhn** [reply]
Got it, thanks
